### ISSUE TITLE
Bundled code is not working. Optional chaining property is renamed when using the same name as export

### ISSUE THREAD

**Issue by ivanStraltsou** [parcel-bundler/parcel#7937]
# 🐛 bug report

bundler renames object properties used via optional chaining and having the same names as exports

## 🎛 Configuration (.babelrc, package.json, cli command)
default getting started template from home page

repo: https://github.com/ivanStraltsou/parcel2-issue
```shell
parcel build ./src/index.html --no-optimize
```

`src/index.html`
```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>My First Parcel App</title>
    <script type="module" src="index.js"></script>
  </head>
  <body>
    <h1>Hello, World!</h1>
  </body>
</html>
```

`src/index.js`
```js
import { isAValue, isBValue } from "./lib";

console.log(isAValue({ isAValue: true }));
console.log(isBValue({ isBValue: true }));
```

`src/lib.js`
```js
// broken
export const isAValue = (payload) => payload?.isAValue === true;

// broken
export const isAAValue = (payload) => payload?.isAValue === true;

// works
export const isAValue = (payload) => payload.isAValue === true;

// works
export const isBValue = (payload) => payload?.bValue === true; 
```

## 🤔 Expected Behavior
result bundle produces valid code, object properties are not renamed

## 😯 Current Behavior
result bundle contains:
```js
const hash$export$hash = (payload)=>payload?.hash$export$hash === true

//---
// broken
const $2e312cb5ec191554$export$33f07d46bc452aa4 = (payload)=>payload?.$2e312cb5ec191554$export$33f07d46bc452aa4 === true
;
const $2e312cb5ec191554$export$3100cf595abe3c86 = (payload)=>payload?.bValue === true
;


console.log($2e312cb5ec191554$export$33f07d46bc452aa4({
    isAValue: true
}));
console.log($2e312cb5ec191554$export$3100cf595abe3c86({
    isBValue: true
}));
;
```
So this value is always undefined and logic is corrupted

## 🔦 Context
- **production code is broken**, hard to capture this type of issues
- using obj?.['property'] style is converted correctly
- using obj.property is converted correctly
- dev server mode is working fine as long as cache is empty
- dev server mode is broken as well when running with cache and bundle first

## 💻 Code Sample
repo: https://github.com/ivanStraltsou/parcel2-issue
```shell
parcel build ./src/index.html --no-optimize
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.4.1
| Node             | 17.9.0 (same on 14.15.4)
| npm/Yarn         | npm 8.5.5 (same on npm 6.14.10)
| Operating System | Win 10 Pro 19043.1586

**Comment by ivanStraltsou** [comment]
seems to be fixed in 2.6.0

**Comment by ivanStraltsou** [comment]
duplicated by https://github.com/parcel-bundler/parcel/issues/8102, fixed in https://github.com/parcel-bundler/parcel/pull/8121
